integration_test/tests/power_bi_tests.rs

```diff
@@ -368,4 +370,165 @@ mod integration {
+    ///         - SQLColAttributeW(SQL_DESC_TYPE_NAME)
+    ///         - SQLColAttributeW(SQL_COLUMN_LENGTH)
+    ///         - SQLColAttributeW(SQL_COLUMN_SCALE)
+    ///     - For X rows or until SQLFetch returns SQL_NO_DATA
+    ///         - SQLFetch()
+    ///         - For columns 1 to {numCols}
+    ///             - SQLGetData({colIndex}, {defaultCtoSqlType})
+    ///     - SQLMoreResults()
+    ///     - SQLFreeHandle(SQL_HANDLE_STMT)
+    #[test]
+    fn test_data_retrieval() {
+        let env_handle: HEnv = setup();
+        let (conn_handle, _, _, _) = power_bi_connect(env_handle);
+        let mut stmt: Handle = null_mut();
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLAllocHandle(
+                    HandleType::Stmt,
+                    conn_handle as *mut _,
+                    &mut stmt as *mut Handle
+                )
+            );
+
+            // SQL_DRIVER_ODBC_VER and SQL_DRIVER_NAME are not available through odbc_sys
+            /*
+            SQLGetInfoW(SQL_DRIVER_ODBC_VER)
+            SQLGetInfoW(SQL_DRIVER_NAME)
+            */
+
+            let mut query: Vec<u16> = "select * from example".encode_utf16().collect();
+            query.push(0);
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLExecDirectW(stmt as HStmt, query.as_ptr(), SQL_NTS as i32),
+                "{}",
+                get_sql_diagnostics(HandleType::Stmt, stmt as Handle)
+            );
+
+            // SQLGetFunctions is not available through odbc_sys
+            /*
+            SQLGetFunctions(SQL_API_SQLFETCHSCROLL)
+            */
+
+            let str_len_ptr = &mut 0;
+            let output_buffer = &mut [0u16; (BUFFER_LENGTH as usize - 1)] as *mut _;
+
+            test_get_info!(
+                conn_handle,
+                InfoType::GetDataExtensions,
+                2,
+                DataType::USmallInt
+            );
+
+            let column_count_ptr = &mut 0;
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLNumResultCols(stmt as HStmt, column_count_ptr)
+            );
+            assert_eq!(2, *column_count_ptr);
+
+            let numeric_attribute_ptr = &mut 0;
+            const FIELD_IDS: [Desc; 7] = [
+                Desc::ConciseType,
+                Desc::Unsigned,
+                Desc::Name,
+                Desc::Nullable,
+                Desc::TypeName,
+                Desc::Length,
+                Desc::Scale,
+            ];
+            for col_num in 0..*column_count_ptr {
+                FIELD_IDS.iter().for_each(|field_type| {
+                    assert_eq!(
+                        SqlReturn::SUCCESS,
+                        SQLColAttributeW(
+                            stmt as HStmt,
+                            (col_num + 1) as u16,
+                            *field_type,
+                            output_buffer as Pointer,
+                            14,
```

Comment:
I couldn't figure out where `14` is form. The actual buffer is length is BUFFER_LENGTH * size_of::<WideChar>, no?

Reply:
I was trying to roughly match how we did `get_info`, though it ended up getting a little messy because we are in a loop (so the outputs have varying sizes -- 14 being the size in bytes of the largest). happy to just make this `BUFFER_SIZE * size_of::` however if that is more clear, since I didn't end up checking the `str_len_ptr`.